docs/demultiplexing.md

```diff
@@ -0,0 +1,171 @@
+        {
+            "name": "my_pipeline_with_demultiplexer",
+            "inputs": ["pipeline_input_name"],
+            "nodes": [
+                {
+                    "name": "Model_A_node",
+                    "model_name": "Model_A",
+                    "type": "DL model",
+                    "demultiply_count": 3,  <------------------ this parameter specifies how many branches should be spawned
+                    "inputs": [
+                        {"input": {"node_name": "request",
+                                   "data_item": "pipeline_input_name"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output_A",
+                         "alias": "output_A"},
+                        {"data_item": "output_B",
+                         "alias": "output_B"}
+                    ]
+                },
+                {
+                    "name": "Model_B_node",
+                    "model_name": "Model_B",
+                    "type": "DL model",
+                    "inputs": [
+                        {"input_A": {"node_name": "Model_A_node",
+                                     "data_item": "output_A"}},
+                        {"input_B": {"node_name": "Model_A_node",
+                                     "data_item": "output_B"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output",
+                         "alias": "output"}
+                    ]
+                }
+            ],
+            "outputs": [
+                {"pipeline_output_name": {"node_name": "Model_B_node",
+                                          "data_item": "output"}}
+            ]
+        }
+    ]
+}
+```
+
+## Dynamic demultiply_count parameter
+There might be use cases where one custom node library is used to produce unpredictable number of batch. To achieve it, `demultiply_count` can be set to `0`. This indicates that pipeline supports any number of batch returned by custom node: `(1,X,N,C,H,W,...)` - where `X` is dynamic `demultiply_count`. OpenVINO&trade; Model Server is capable of interpreting such dynamic batch and is able to split outputs into dynamic number of pipeline branches. When using dynamic `demultiply_count` parameters, only one demultiplexer can exist in pipeline. Important to note - when batch 0 is returned, pipeline stops its execution and reports an error to gRPC/REST response.
+
+## Multiple demultiplexers
+Directed Acyclic Graph Scheduler is not limited to single demultiplexer node in one pipeline definition. Each demultiplexer node which has no manually specified gather node will be automatically gathered in `response` node - meaning each demultiplexer adds one new dimension equal to `demultiply_count` into all pipeline outputs shape. This must be taken into account when interpreting response data in client applications.
+
+![diagram](multiple_demultiplexers.svg)
+
+## Manual gathering
```

Comment:
manually?

Reply:
Configuring gathering?

Reply:
_Default_ word would fit better

Reply:
Add example use case

Reply:
Done - got rid of `manual` word completely